[API-603] Update hazelcast.org links

yemreinci opened the pull request:
Fixes #894.

ihsandemir commented:
Not related to this PR (it can be a separate PR) but the links are not working at these sentences:

1. `See the advanced installation section for configuration options.`
2. `See this section for information on how to use a different installation location.`
3. `Hazelcast offers its own native serialization types, such as identified_data_serializer serialization and portable_serializer serialization.`
4. `See the C++ Client Operation Modes section for the description of smart and unisocket modes.`
5. `You can create a Hazelcast cluster in the cloud with "Enable Encryption" option. `
6. `You can configure connection_attempt_limit for the number of times you want the client to retry connecting. See the Setting Connection Attempt Limit section.`
7. `For adding your own compiled class or JAR files to the server's CLASSPATH, see the Adding User Library to CLASSPATH section.`


boost asio links refer to version 1.75 can be changed to 1.76.

devOpsHazelcast commented:
Linux test PASSed.

yemreinci commented:
> Not related to this PR (it can be a separate PR) but the links are not working at these sentences:
> 
>     1. `See the advanced installation section for configuration options.`
> 
>     2. `See this section for information on how to use a different installation location.`
> 
>     3. `Hazelcast offers its own native serialization types, such as identified_data_serializer serialization and portable_serializer serialization.`
> 
>     4. `See the C++ Client Operation Modes section for the description of smart and unisocket modes.`
> 
>     5. `You can create a Hazelcast cluster in the cloud with "Enable Encryption" option. `
> 
>     6. `You can configure connection_attempt_limit for the number of times you want the client to retry connecting. See the Setting Connection Attempt Limit section.`
> 
>     7. `For adding your own compiled class or JAR files to the server's CLASSPATH, see the Adding User Library to CLASSPATH section.`
> 
> 
> boost asio links refer to version 1.75 can be changed to 1.76.

Addressed all except 6 in https://github.com/hazelcast/hazelcast-cpp-client/pull/920/commits/5dedbd8b354303025c8599d0412dbd852d7c90c8. 

I couldn't find a `connection_attempt_limit` configuration or any related doc section.

devOpsHazelcast commented:
Linux test PASSed.

devOpsHazelcast commented:
Linux test PASSed.

devOpsHazelcast commented:
Windows test PASSed.

devOpsHazelcast commented:
Linux test PASSed.

devOpsHazelcast commented:
Windows test PASSed.

devOpsHazelcast commented:
Windows test PASSed.

devOpsHazelcast commented:
Windows test PASSed.

ihsandemir commented:
> connection_attempt_limit

Because, it changed to `## 5.5. Advanced Cluster Connection Retry Configuration` now. that was old config. That section needs to be re-worded to use the `connection_retry_config`.

ihsandemir commented on Reference_Manual.md:
```diff
@@ -812,7 +812,7 @@ Hazelcast serializes all your objects before sending them to the server. The `un
 Vector of the above types can be serialized as `boolean[]`, `byte[]`, `short[]`, `int[]`, `float[]`, `double[]`, `long[]` and `string[]` for the Java server side, respectively. 
 
-If you want the serialization to work faster or you use the clients in different languages, Hazelcast offers its own native serialization types, such as [`identified_data_serializer` serialization](#41-identifieddataserializable-serialization) and [`portable_serializer` serialization](#42-portable-serialization).
+If you want the serialization to work faster or you use the clients in different languages, Hazelcast offers its own native serialization types, such as [`identified_data_serializer` serialization](#41-identified_data_serializer-serialization) and [`portable_serializer` serialization](#42-portable-serialization).
```
```suggestion
If you want the serialization to work faster or you use the clients in different languages, Hazelcast offers its own native serialization types, such as [`identified_data_serializer` serialization](#41-identified_data_serializer-serialization) and [`portable_serializer` serialization](#42-portable_serializer-serialization).
```

ihsandemir reviewed (commented):
Link still broken:
 `and portable_serializer serialization.`

yemreinci commented:
> > connection_attempt_limit
> 
> Because, it changed to `## 5.5. Advanced Cluster Connection Retry Configuration` now. that was old config. That section needs to be re-worded to use the `connection_retry_config`.

OK, I updated the section to be similar to [Java client's doc](https://docs.hazelcast.com/imdg/4.2/clients/java.html#handling-failures). Commit: https://github.com/hazelcast/hazelcast-cpp-client/pull/920/commits/c3a6b7f114f81e01a2eb2b817440adcb22c850cd

Also, fixed more broken links and some other problems in https://github.com/hazelcast/hazelcast-cpp-client/pull/920/commits/94e02fdf256ee858a299d87a40575d889dfe3eee

devOpsHazelcast commented:
Windows test PASSed.

devOpsHazelcast commented:
Linux test PASSed.

devOpsHazelcast commented:
Linux test FAILed.

devOpsHazelcast commented:
Linux test FAILed.

devOpsHazelcast commented:
Linux test FAILed.

devOpsHazelcast commented:
Windows test FAILed.

devOpsHazelcast commented:
Windows test FAILed.

devOpsHazelcast commented:
Windows test FAILed.

devOpsHazelcast commented:
Linux test PASSed.

devOpsHazelcast commented:
Windows test PASSed.

devOpsHazelcast commented:
Linux test PASSed.

devOpsHazelcast commented:
Windows test PASSed.
